components/servo/proxies.rs

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+use compositing_traits::ConstellationMsg;
+use crossbeam_channel::{SendError, Sender};
+
+#[derive(Clone)]
+pub(crate) struct ConstellationProxy {
+    sender: Sender<ConstellationMsg>,
+    disconnected: Arc<AtomicBool>,
+}
+
+impl ConstellationProxy {
+    pub fn new(sender: Sender<ConstellationMsg>) -> Self {
+        Self {
+            sender,
+            disconnected: Arc::default(),
+        }
+    }
+
+    pub fn send(&self, msg: ConstellationMsg) {
+        let _ = self.try_send(msg);
```

Comment:
Could we add error log here if it fails?

Reply:
The idea here is that we will not deliver a warning here. Instead there will be a delegate for the `Servo` object itself. When we lose the connection to `Servo`, instead of printing a warning we will give call an error method on the delegate so that the application can shut down properly. In that case it could also print a warning. 

That said, there's no issue with also printing a warning here, I think.